savoirfairelinux/sous-chef
# Fixed #627: Node.js packages are outdated

## Fixes #627 by @lingxiaoyang 

### Changes proposed in this pull request:

* Update the versions of all JS dependencies (both frontend and for Gulp)
* Import vendor JS/CSS/images from npm packages, rather than having them hard-coded into this repository.
* Set the Docker environment to use Node.JS 6.x (LTS) versions, instead of 4.x.

### Status


### How to verify this change

* Rebuild the Docker dev environment using `docker-compose build && docker-compose up`.
* Launch the Docker dev environment using `docker exec -it $IDOFWEBCONTAINER bash`.
* Run `node --version && npm --version`
* Ensure the new npm dependencies install properly with `cd /code/tools/gulp && npm install`
* Ensure the Gulp pipeline runs successfully with `cd /code/tools/gulp && gulp watch`
* Open the dev server at `http://localhost:8000` and ensure all assets are properly loading on various screens.

### Deployment notes and migration

- [X] Environment rebuild is needed for this change

It is imperative to **rebuild your docker-compose environment** using `docker-compose build` so the new repos for Node.JS 6.x can be properly installed.

### New translatable strings


### Additional notes

none
